src/magneticField/CMZField.cpp

```diff
@@ -0,0 +1,441 @@
+		A=(1.+6.*(r-rr)/(2.*r-3.*rr) * ( tan(m/eta *log((r+1.)/(R+1.)) +m*phi) - sin(m/eta *log((r+1.)/(R+1.)))/cos(m/eta *log((r+1.)/(R+1.)) +m*phi) )) ;
+	}
+	else
+	{
+    	A= 1.;
+	}
+	return -1/eta *r/(r+1.) * BrAz(r,phi,z,m,B1,eta,R,rr) * A;
+}
+
+// azimuthal field in cartesian  coordinates// for molecular cloud region
+double CMZField::BxAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (cos(phi)-1./eta * sin(phi)) *BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+double CMZField::ByAz(double x, double y, double z, double m, double B1, double eta, double R, double rr) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    return (sin(phi)+1./eta*cos(phi))*BrAz(r,phi,z,m,B1,eta,R,rr);
+}
+
+// Polodial field in cartesian coordinates// for non-thermal filament and the intercloud region
+double CMZField::ByPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;// B1 is the observed magnetic field strength and eta the normalization factor otained from the integrations due to the normalization
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return sin(phi)*Br(r,z,B2,a,L);
+}
+
+double CMZField::BxPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return cos(phi)*Br(r,z,B2,a,L);
+}
+
+double CMZField::BzPol(double x, double y, double z, double B1, double a1, double a2, double eta) const{
+    double r= sqrt(pow(x,2.)+pow(y,2.));
+    double phi= std::atan2(y,x);
+    double B2=B1/eta;
+    double L=scale(a2/2.,1);
+    double a= 1/pow(scale(a1/2.,2),2.);
+    return Bz(r,z,B2,a,L);
+}
+/////end model C + azimuthal component
+
+bool CMZField::getUseMCField() const {
+    return useMCField;
+}
+bool CMZField::getUseICField() const {
+    return useICField;
+}
+bool CMZField::getUseNTFField() const {
+    return useNTFField;
+}
+bool CMZField::getUseRadioArc() const {
+    return useRadioArc;
+}
+
+void CMZField::setUseMCField(bool use) {
+    useMCField = use;
+}
+void CMZField::setUseICField(bool use) {
+    useICField = use;
+}
+void CMZField::setUseNTFField(bool use) {
+    useNTFField = use;
+}
+void CMZField::setUseRadioArc(bool use) {
+    useRadioArc = use;
+}
+
+// Parameter identification:
+// r: radius in cylindrical coordinates
+// z= z component in cylindircal coordintes
+// phi: phi component in cylindircal coordinates
+// B1: observed magnetic field strength
+// B2: normalization magnetic field strength according to B1
+// m: azimuthal wavenumber m=0 for axissymmetric, m=1 for bisymmetric and m=2 quadrosymmetric, ... field lines
```

Comment:
m: azimuthal wavenumber (m=0 for axisymmetric, m=1 for bisymmetric, etc field lines)